cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/KafkaClusterState.java

```diff
@@ -114,376 +97,412 @@ protected int minInsyncReplicas(String topic) {
-    Pattern topic = kafkaClusterStateParams.topic();
-    boolean verbose = kafkaClusterStateParams.isVerbose();
-
-    String initMessage = verbose ? "All Partitions in the Cluster (verbose):"
-                                 : "Under Replicated, Offline, and Under MinIsr Partitions:";
-    sb.append(String.format("%n%s%n%" + topicNameLength + "s%10s%10s%30s%30s%25s%25s%n", initMessage, "TOPIC",
-                            "PARTITION", "LEADER", "REPLICAS", "IN-SYNC", "OUT-OF-SYNC", "OFFLINE"));
-
-    // Gather the cluster state.
-    Comparator<PartitionInfo> comparator =
-        Comparator.comparing(PartitionInfo::topic).thenComparingInt(PartitionInfo::partition);
-    SortedSet<PartitionInfo> underReplicatedPartitions = new TreeSet<>(comparator);
-    SortedSet<PartitionInfo> offlinePartitions = new TreeSet<>(comparator);
-    SortedSet<PartitionInfo> otherPartitions = new TreeSet<>(comparator);
-    SortedSet<PartitionInfo> partitionsWithOfflineReplicas = new TreeSet<>(comparator);
-    SortedSet<PartitionInfo> underMinIsrPartitions = new TreeSet<>(comparator);
-
-    populateKafkaPartitionState(underReplicatedPartitions,
-                                offlinePartitions,
-                                otherPartitions,
-                                partitionsWithOfflineReplicas,
-                                underMinIsrPartitions,
-                                verbose,
-                                topic);
-
-    // Write the cluster state.
-    sb.append(String.format("Offline Partitions:%n"));
-    writeKafkaPartitionState(sb, offlinePartitions, topicNameLength);
-
-    sb.append(String.format("Partitions with Offline Replicas:%n"));
-    writeKafkaPartitionState(sb, partitionsWithOfflineReplicas, topicNameLength);
-
-    sb.append(String.format("Under Replicated Partitions:%n"));
-    writeKafkaPartitionState(sb, underReplicatedPartitions, topicNameLength);
-
-    sb.append(String.format("Under MinIsr Partitions:%n"));
-    writeKafkaPartitionState(sb, underMinIsrPartitions, topicNameLength);
-
-    if (verbose) {
-      sb.append(String.format("Other Partitions:%n"));
-      writeKafkaPartitionState(sb, otherPartitions, topicNameLength);
+    protected void writeKafkaPartitionState(StringBuilder sb, Set<PartitionInfo> partitions, int topicNameLength) {
+      for (PartitionInfo partitionInfo : partitions) {
+        sb.append(new PartitionState(partitionInfo).writeKafkaPartitionState(topicNameLength));
+      }
     }
   }
 
-  protected String getPlaintext(CruiseControlParameters parameters) {
-    StringBuilder sb = new StringBuilder();
-    try {
-      // Broker summary.
-      writeBrokerSummary(sb);
-      // Partition summary.
-      writePartitionSummary(sb, parameters);
-    } catch (InterruptedException | ExecutionException e) {
-      throw new RuntimeException("Failed to populate broker logDir state.", e);
+
+  @JsonResponseClass
+  protected static class PartitionState {
```

Comment:
Nit on style: I am curious about why we prefer using a static inner class as opposed to a separate class.